recipes/lodepng/CMakeLists.txt

```
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.23)
+
+include(GNUInstallDirs)
+
+project(lodepng)
```

```suggestion
project(lodepng
        VERSION "$ENV{PKG_VERSION}"
        LANGUAGES CXX)
```
People may appreciate having the API version in the CMake metadata.

---

I added the change, but this is not sufficient on its own to make sure that the version is available to downstream CMake users. We also need to add a call to [write_basic_package_version](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#command:write_basic_package_version_file) and install the generated file.

---

Completed the support in https://github.com/conda-forge/staged-recipes/pull/19104/commits/421c269b1e5af6af5caea818f0f20e8eed84d66c .